pkg/yurtmanager/controller/loadbalancerset/viploadbalancer/viploadbalancer_controller.go

```diff
@@ -0,0 +1,290 @@
+	}
+
+	return nil
+}
+
+func (r *ReconcileVipLoadBalancer) syncVRID(ctx context.Context, poolService *netv1alpha1.PoolService) error {
+	currentVRIDs, err := r.getCurrentVRID(ctx, poolService)
+	if err != nil {
+		return fmt.Errorf("failed to get current PoolServices: %v", err)
+	}
+
+	r.VRIDManager.SyncVRID(poolService.Labels[network.LabelNodePoolName], currentVRIDs)
+	return nil
+}
+
+func (r *ReconcileVipLoadBalancer) getCurrentVRID(ctx context.Context, poolService *netv1alpha1.PoolService) ([]int, error) {
+	// Get the poolservice list
+	listSelector := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			network.LabelNodePoolName: poolService.Labels[network.LabelNodePoolName],
+		}),
+		FieldSelector: fields.SelectorFromSet(map[string]string{
+			"spec.loadBalancerClass": VipLoadBalancerClass,
+		}),
+	}
+	poolServiceList := &netv1alpha1.PoolServiceList{}
+	if err := r.List(ctx, poolServiceList, listSelector); err != nil {
+		return nil, err
+	}
+
+	return filterInvalidPoolService(poolServiceList.Items), nil
+}
+
+func filterInvalidPoolService(poolServices []netv1alpha1.PoolService) []int {
+	poolVrids := []int{}
+	for _, poolService := range poolServices {
+		if poolService.Labels == nil {
+			continue
+		}
+
+		if _, ok := poolService.Annotations[AnnotationVipLoadBalancerVRID]; !ok {
+			continue
+		}
+
+		vrid, err := strconv.Atoi(poolService.Annotations[AnnotationVipLoadBalancerVRID])
+		if err != nil {
+			klog.Errorf(Format("Failed to convert VRID to int: %v", err))
+			continue
+		}
+
+		if vrid < MINVRIDLIMIT || vrid > MAXVRIDLIMIT {
+			klog.Errorf(Format("Invalid VRID %d", vrid))
+			continue
+		}
+
+		poolVrids = append(poolVrids, vrid)
+	}
+
+	return poolVrids
+}
+
+func (r *ReconcileVipLoadBalancer) hasValidVRID(poolService netv1alpha1.PoolService) bool {
+	if poolService.Annotations == nil {
+		return false
+	}
+
+	if _, ok := poolService.Annotations[AnnotationVipLoadBalancerVRID]; !ok {
+		return false
+	}
+
+	return true
+}
+
+func (r *ReconcileVipLoadBalancer) handleVRID(poolService *netv1alpha1.PoolService) error {
+	if r.hasValidVRID(*poolService) {
+		return nil
+	}
+
+	// Assign a new VRID to the PoolService
+	if err := r.assignVRID(poolService); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileVipLoadBalancer) assignVRID(poolService *netv1alpha1.PoolService) error {
+	// Get the poolName from the PoolService
+	poolName := poolService.Labels[network.LabelNodePoolName]
+	// Get a new VRID
+	vrid := r.VRIDManager.GetVRID(poolName)
+	if vrid == EVICTED {
+		return fmt.Errorf("VRID usage limit exceeded")
+	}
+
+	// Set the VRID to the PoolService
+	if poolService.Annotations == nil {
+		poolService.Annotations = make(map[string]string)
+	}
+
+	poolService.Annotations[AnnotationVipLoadBalancerVRID] = strconv.Itoa(vrid)
+	return nil
+}
+
+func (r *ReconcileVipLoadBalancer) reconcileDelete(ctx context.Context, poolService *netv1alpha1.PoolService) (reconcile.Result, error) {
```

Comment:
When a poolService is deleted, loadBalancerSet controller only setup the `DeletionTimestamp`, then `vipLoadBalancer` should cleanup all resources of LoadBalancer in this NodePool and remove `finalizer` in the poolService.

Reply:
there is no need for `vipLoadBalancer` controller to delete `poolService` resource, and only need to remove the finalizer after all resources of loadbalancer on edge nodes are cleaned up.